config/routes.rb

```diff
@@ -0,0 +1,18 @@
+Rails.application.routes.draw do
+  # For details on the DSL available within this file, see http://guides.rubyonrails.org/routing.html
+  root 'rideshare#index'
+
+  resources :rideshare, only:[:index]
+
+  resources :drivers do
+    member do
+      patch :update_status
+    end
+  end
+
+  resources :trips, only: [:show, :update]
+
+  resources :passengers do
+    resources :trips, only: [:create, :show]
+  end
```

Comment:
I don't think you need the `:show` route for trips nested under passengers. That would create a route like `passengers/:passenger_id/trips/:id`.